tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,431 @@
+        for the coordinate variables
+        """
+        prefetch_dataset = Dataset(self.nc4file, 'r')
+        expected_shape = (406, 964)
+        with self.subTest('Expected lat/lon arrays'):
+            lat_prefetch_arr, lon_prefetch_arr = get_lat_lon_arrays(
+                prefetch_dataset,
+                self.varinfo.get_variable(self.latitude),
+                self.varinfo.get_variable(self.longitude),
+            )
+            self.assertTupleEqual(lat_prefetch_arr.shape, expected_shape)
+            self.assertTupleEqual(lon_prefetch_arr.shape, expected_shape)
+
+        with self.subTest('Missing coordinate'):
+            with self.assertRaises(MissingCoordinateVariable) as context:
+                lat_prefetch_arr, lon_prefetch_arr = get_lat_lon_arrays(
+                    prefetch_dataset,
+                    self.varinfo.get_variable(
+                        '/Soil_Moisture_Retrieval_Data_AM/latitude_centroid'
+                    ),
+                    self.varinfo.get_variable(
+                        '/Soil_Moisture_Retrieval_Data_AM/longitude_centroid'
+                    ),
+                )
+                self.assertEqual(
+                    context.exception.message,
+                    'Coordinate: "/Soil_Moisture_Retrieval_Data_AM/latitude_centroid" is '
+                    'not present in source granule file.',
+                )
+
+    def test_get_override_projected_dimension_names(self):
+        """Ensure that the expected projected dimension name
+        is returned for the coordinate variables
+        """
+
+        expected_projected_names = [
+            '/Soil_Moisture_Retrieval_Data_AM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_AM/projected_x',
+        ]
+
+        with self.subTest(
+            'Retrieves expected projected dimension names for a science variable'
+        ):
+            self.assertListEqual(
+                get_override_projected_dimension_names(self.varinfo, self.latitude),
+                expected_projected_names,
+            )
+
+        with self.subTest(
+            'Retrieves expected dimension names for the longitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimension_names(self.varinfo, self.longitude),
+                expected_projected_names,
+            )
+
+        with self.subTest('Raises exception for missing coordinate variable'):
+            with self.assertRaises(MissingCoordinateVariable) as context:
+                get_override_projected_dimension_names(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_AM/longitude_centroid'
+                )
+                self.assertEqual(
+                    context.exception.message,
+                    'Coordinate: "/Soil_Moisture_Retrieval_Data_AM/latitude_centroid" is '
+                    'not present in source granule file.',
+                )
+
+    def test_get_override_projected_dimensions(self):
```

Comment:
Another great set of unit tests. Again, I could only think of one more: when `latitude_coordinates` and `longitude_coordinates` as found inside the function are both empty lists, and the overall output is also just an empty list.

Reply:
I don't think I have a real test case for it. I may have to use a fake dmr if we need to do that